app/src/main/java/org/mozilla/focus/shortcut/IconGenerator.kt

```diff
@@ -23,37 +23,55 @@ class IconGenerator {
 
     companion object {
         private val TEXT_SIZE_DP = 36f
-        private val DEFAULT_ICON_CHAR = "?"
+        private val DEFAULT_ICON_CHAR = '?'
 
         /**
-         * Use the given raw website icon and generate a launcher icon from it. If the given icon is null
-         * or too small then an icon will be generated based on the website's URL. The icon will be drawn
+         * Generate an icon based on the website's URL. The icon will be drawn
          * on top of a generic launcher icon shape that we provide.
          */
         @JvmStatic
-        fun generateLauncherIcon(context: Context, url: String?) = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            generateAdaptiveLauncherIcon(context, url)
+        fun generateLauncherIcon(context: Context, url: String?): Bitmap {
+            val startingChar = getRepresentativeCharacter(url)
+            return generateCharacterIcon(context, startingChar)
+        }
+
+        /**
+         * Generate an icon with the given character. The icon will be drawn
+         * on top of a generic launcher icon shape that we provide.
+         */
+        @JvmStatic
+        fun generateCharacterIcon(context: Context, character: Char) = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            generateAdaptiveLauncherIcon(context, character)
         } else {
-            generateLauncherIconPreOreo(context, url)
+            generateLauncherIconPreOreo(context, character)
+        }
+
+        @JvmStatic
+        fun generateLauncherIconPreOreo(context: Context, url: String?): Bitmap {
```

Comment:
I'd rather expose `getRepresentativeCharacter` to the code calling this and make it use the Char method than to duplicate the code